luxonis_ml/nn_archive/config_building_blocks/enums/image_layout.py

```diff
@@ -0,0 +1,10 @@
+from enum import Enum
+
+
+class ImageLayout(Enum):
+    """Represents the input image layout of the model."""
+
```

Comment:
Do we want to include batch dimension in the layout?

Reply:
I was thinking it might be easier for users to decide if we have both batched and non-batched options available (they might get lost if their models has 4 input dimensions while we only provide 3 here). But it's true that ignoring the batch dimension is more clean. Thus, I'm removing the batch dimension options in [d0652e9](https://github.com/luxonis/luxonis-ml/pull/109/commits/d0652e97619e97eaacd1432f729e71426587841e)

Reply:
Sorry, should have been more clear. I meant if we should enforce batch dimension on all inputs?

Reply:
I see. If that's desirable, we can force it by an additional `model_validator` to the `Input` class doing the checks on `input_type` and `shape` fields - if `InputType.IMAGE`, then `shape` must be of length=4.

Reply:
Added in [2fd6907](https://github.com/luxonis/luxonis-ml/pull/109/commits/2fd69074e8e55c4f160d8f28b5a9b1042a93f3a5).